### PR TITLE
flottbot: init at 0.11.0

### PR DESCRIPTION
**Pull request by bryanhonof**
###### Description of changes

Initialize [flottbot] at version 0.11.0

[flottbot]: https://github.com/target/flottbot

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 229444` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>flottbot</li>
  </ul>
</details>

File: pkgs/applications/networking/flottbot/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, buildGoModule, fetchFromGitHub, nixosTests, nix-update-script }:
+buildGoModule rec {
+  pname = "flottbot";
+  version = "0.11.0";
+
+  src = fetchFromGitHub {
+    owner = "target";
+    repo = pname;
+    rev = "${pname}-${version}";
```
Comment (Janik-Haag) [inline pkgs/applications/networking/flottbot/default.nix:9]: Currently the url syntax is wrong, you only have to add the version in this case ande the wrapper does the rest.
You usally give the hash of a commit or the version number of a tag, you can look at other packages using the github wrapper with: ```rg fetchFromGitHub``` to get some examples
```suggestion
    rev = version;
```

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/flottbot/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "538c2SkyOYmm84yM0svbPeJtSIMHRI6mot8MIP5wBfc=";
+  };
+
+  vendorSha256 = "sha256-6j1nQBlFqd78v/CNV7ivwYZ5CEzGh+An6xcQr4dAkAs=";
+
+  subPackages = [ ];
+
+  doCheck = false; # Tries to do some networking :(
```
Comment (Janik-Haag) [inline pkgs/applications/networking/flottbot/default.nix:17]: If I set this to true the test run just fine o.O

```
nix-build -A flottbot
this derivation will be built:
  /nix/store/h5m8q5j99nsnp7lp9xgpqzijvgcspnid-flottbot-0.12.0.drv
building '/nix/store/h5m8q5j99nsnp7lp9xgpqzijvgcspnid-flottbot-0.12.0.drv'...
unpacking sources
unpacking source archive /nix/store/wk3cahkfvxwklcbdham4xf8gnr5jcsc2-source
source root is source
patching sources
configuring
building
Building subPackage ./cmd/flottbot
Building subPackage ./config-example/scripts
Building subPackage ./core
Building subPackage ./handlers
Building subPackage ./models
Building subPackage ./remote
Building subPackage ./remote/cli
Building subPackage ./remote/discord
Building subPackage ./remote/gchat
Building subPackage ./remote/scheduler
Building subPackage ./remote/slack
Building subPackage ./remote/telegram
Building subPackage ./utils
Building subPackage ./version
buildPhase completed in 38 seconds
running tests
ok      github.com/target/flottbot/core 2.035s
ok      github.com/target/flottbot/handlers     1.018s
ok      github.com/target/flottbot/utils        0.010s
ok      github.com/target/flottbot/version      0.003s
checkPhase completed in 33 seconds
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0
shrinking /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0/bin/scripts
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
shrinking /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0/bin/flottbot
checking for references to /build/ in /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
patching script interpreter paths in /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0
stripping (with command strip and flags -S) in  /nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0/bin
/nix/store/2g1p03mqx4kjr668zj9xhx3jnnmlk750-flottbot-0.12.0
```

which test did you have prolems with?

File: pkgs/applications/networking/flottbot/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{ lib, buildGoModule, fetchFromGitHub, nixosTests, nix-update-script }:
```
Comment (Janik-Haag) [inline pkgs/applications/networking/flottbot/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, nix-update-script
, nixosTests
}:
```

File: pkgs/applications/networking/flottbot/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "target";
+    repo = pname;
+    rev = version;
+    sha256 = "538c2SkyOYmm84yM0svbPeJtSIMHRI6mot8MIP5wBfc=";
+  };
+
+  vendorSha256 = "sha256-6j1nQBlFqd78v/CNV7ivwYZ5CEzGh+An6xcQr4dAkAs=";
+
+  subPackages = [ ];
```
Comment (Janik-Haag) [inline pkgs/applications/networking/flottbot/default.nix:15]: What do you need this for?

**Review by Janik-Haag — COMMENTED** [review]
In there is `./result/bin/scripts` which just prints `Hi from Go` which seems like it shouldn't be there.

File: pkgs/applications/networking/flottbot/default.nix (hunk 2 above)
Comment (bryanhonof) [reply pkgs/applications/networking/flottbot/default.nix:17]: Don't really remember. Maybe they removed/fixed that test case in `0.12.0`. I'll try agian.

File: pkgs/applications/networking/flottbot/default.nix (hunk 4 above)
Comment (bryanhonof) [reply pkgs/applications/networking/flottbot/default.nix:20]: Probably copy/paste from another Go package, sorry. 😅

Comment (Janik-Haag) [reply pkgs/applications/networking/flottbot/default.nix:20]: Could you delete it?

File: pkgs/applications/networking/flottbot/default.nix (hunk 2 above)
Comment (Janik-Haag) [reply pkgs/applications/networking/flottbot/default.nix:22]: Sure

**Review by Janik-Haag — COMMENTED** [review]
This should be good to go after addressing the current discussions and squashing commits so we only have the `flottbot: init at 0.11.0` commit.

**Comment by bryanhonof** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/290084

**Pull request closed without merging**